lib/settings/settings_service.dart

```diff
@@ -83,6 +88,17 @@ class SettingsService {
     );
   }
 
+  Future<void> setLastFmAuth() async {
```

Comment:
ah, sorry maybe I was a bit too vague. this needs to be inside the exposeservice ofc not in the settingsservice
the settingsservice only does settings and nothing else
then I guess you need an expose model or put it into the app model

Reply:
I cannot add a dependency of settingsmodel to exposeservice, so I moved the storing of api keys to app model. That is OK?

Reply:
that's fine, great work, thank you, I think the only thing left is we would need to move away from future delayed and user a timer instead, which you need to cancel in the dispose in the end but it can be re-used